Makefile

```diff
@@ -66,8 +66,37 @@ coveralls: $(COV_REPORT)
 
 .PHONY: bench
 BENCH ?= .
+BENCH_FILE ?= .bench/new.txt
 bench:
-	$(ECHO_V)$(foreach pkg,$(BENCH_PKGS),go test -bench=$(BENCH) -run="^$$" $(BENCH_FLAGS) $(pkg);)
+	$(ECHO_V)$(foreach pkg,$(BENCH_PKGS),go test -bench=. -run="^$$" $(BENCH_FLAGS) $(pkg) |\
+		tee $(BENCH_FILE);)
+
+.PHONY: benchbase
+BASELINE_BENCH_FILE=.bench/old.txt
+benchbase:
+	$(ECHO_V)if [ -z "$(IGNORE_BASELINE_CHECK)" ] && [ -z "$(git diff master)" ]; then \
+		echo "$(ERROR_STYLE)Can't record baseline with code changes off master." ; \
+		echo "Check out master and try again$(COLOR_RESET)"; \
+		exit 1; \
+	fi
+
+	@echo "$(LABEL_STYLE)Running baseline benchmark$(COLOR_RESET)"
+	@echo
+	$(ECHO_V)$(MAKE) bench BENCH_FILE=$(BASELINE_BENCH_FILE)
+
+.PHONY: benchcmp
+benchcmp:
+	$(ECHO_V)which benchcmp >/dev/null || go get -u golang.org/x/tools/cmd/benchcmp
+	$(ECHO_V)test -s $(BASELINE_BENCH_FILE) ||\
+		$(error "Must have a baseline bench file. *Check out master* and run `make benchbase`")
+	$(ECHO_V)test -s $(BENCH_FILE) || $(error "Must have a new benchmark file. Run `make bench`")
+	$(ECHO_V)benchcmp $(BASELINE_BENCH_FILE) $(BENCH_FILE)
+
+.PHONY: benchclean
+benchclean:
+	$(ECHO_V)if [ -a $(BASELINE_BENCH_FILE) ]; then rm $(BASELINE_BENCH_FILE); fi
```

Comment:
-a
file exists

This is identical in effect to -e. It has been "deprecated," [1] and its use is discouraged.

http://www.tldp.org/LDP/abs/html/fto.html

Also just do rm -f $(BASELINE_BENCH_FILE) and don't bother checking it first.